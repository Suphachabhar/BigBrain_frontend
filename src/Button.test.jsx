import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './components/Button.styles';

describe('Button', () => {
  const noop = () => {};

  // test the button component
  // it = particular object -> Button
  it('trigger onClick event handler when clicked', () => {
    const onClick = jest.fn();
    shallow(<Button onClick="{onClick}" />).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('uses custom title', () => {
    const title = 'A custom title';
    const button = shallow(<Button onClick={noop} title={title} />);
    expect(button.text()).toBe(title);
  });

  it('renders with minimal props', () => {
    const button = renderer.create(<Button onClick={noop} />).toJSON;
    expect(button).toMatchSnapshot();
  });

  it('renders with provided title (snapshot)', () => {
    const button = renderer.create(<Button onClick={noop} title={'title'} />).toJSON;
    expect(button).toMatchSnapshot();
  });
});
