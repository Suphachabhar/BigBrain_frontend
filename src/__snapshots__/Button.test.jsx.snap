// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with minimal props 1`] = `[Function]`;

exports[`Button renders with provided title (snapshot) 1`] = `[Function]`;
